### PR TITLE
Add get include and include headers

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
@PokhodenkoSA @diptorupd 

1.  Headers are included in the layout of the package.
2.  DPPL*Interface shared libraries are moved from environment/lib to the dpctl site-package location (Thanks @Alexander-Makaryev for pointing out $ORIGIN magic variable)
3.  Introduced `dpctl.get_include()`
4. Added `dpctl/__init__.pxd`, which allows downstream users to do `cimport dpctl`.

### CODE REVIEW

File: conda-recipe/bld.bat [hunk 1]
```diff
@@ -25,17 +28,23 @@ ninja install
...
 cd ..
+xcopy install\lib\*.lib dpctl /E /Y
+xcopy install\lib\*.dll dpctl /E /Y
+
+mkdir dpctl\include
+xcopy backends\include dpctl\include /E /Y
+
+
 
 REM required by dpglue
```
Comment (diptorupd) [inline conda-recipe/bld.bat:39]: required by _opencl_core (dpctl.ocldrv)

Comment (diptorupd) [reply conda-recipe/bld.bat:39]: Since we are touching the file, might as well fix the comment.

File: conda-recipe/bld.bat [hunk 2]
```diff
@@ -25,17 +28,23 @@ ninja install
...
-set "DPPL_OPENCL_INTERFACE_LIBDIR=%LIBRARY_PREFIX%/lib"
-set "DPPL_OPENCL_INTERFACE_INCLDIR=%LIBRARY_PREFIX%/include"
-set "OpenCL_LIBDIR=%DPCPP_ROOT%/lib"
+set "DPPL_OPENCL_INTERFACE_LIBDIR=dpctl"
+set "DPPL_OPENCL_INTERFACE_INCLDIR=dpctl\include"
+set "OpenCL_LIBDIR=%DPCPP_ROOT%\lib"
 
 REM required by oneapi_interface
```
Comment (diptorupd) [inline conda-recipe/bld.bat:44]: required by _sycl_core (dpct)

File: conda-recipe/build.sh [hunk 3]
```diff
@@ -22,31 +25,34 @@ DPCPP_ROOT=${ONEAPI_ROOT}/compiler/latest/linux/
...
+mkdir -p dpctl/include
+cp -r backends/include/* dpctl/include
 
 # required by dpctl.opencl_core
-export DPPL_OPENCL_INTERFACE_LIBDIR=${PREFIX}
-export DPPL_OPENCL_INTERFACE_INCLDIR=${PREFIX}/include
-export OpenCL_LIBDIR=${DPCPP_ROOT}/lib
+export DPPL_OPENCL_INTERFACE_LIBDIR=dpctl
+export DPPL_OPENCL_INTERFACE_INCLDIR=dpctl/include
+export OpenCL_LIBDIR=${DPCPP_ROOT}/libw
```
Comment (diptorupd) [inline conda-recipe/build.sh:46]: what is `libw`

File: MANIFEST.in [hunk 4]
```diff
@@ -1 +1,5 @@
 include versioneer.py
+recursive-include dpctl/include *.h *.hpp
+include dpctl/*.pxd
+include dpctl/*DPPL*Interface.*
+global-exclude *.cpp
```
Comment (diptorupd) [inline MANIFEST.in:5]: new line

**Review by diptorupd — COMMENTED** [review]
Pointed out a few cosmetic changes. Will defer to Sergey on how this works out inside CI.

**Comment by diptorupd** [comment]
I am getting an AssertFailure when I do `conda build conda-recipe` 

> Packaging dpctl
> INFO:conda_build.build:Packaging dpctl
> INFO conda_build.build:build(2222): Packaging dpctl
> /home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/environ.py:452: UserWarning: The environment variable 'ONEAPI_ROOT' is being passed through with value '/opt/intel/oneapi'.  If you are splitting build and test phases with --no-test, please ensure that this value is also set similarly at test time.
>   UserWarning
> Packaging dpctl-0.3.0-py36h6bb024c_5
> INFO:conda_build.build:Packaging dpctl-0.3.0-py36h6bb024c_5
> INFO conda_build.build:bundle_conda(1468): Packaging dpctl-0.3.0-py36h6bb024c_5
> compiling .pyc files...
> found egg dir: /home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/dpctl-0.3.0+5.gbcf97e2-py3.6-linux-x86_64.egg
> number of files: 52
> WARNING :: get_rpaths_raw()=[] and patchelf=[['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib', '$ORIGIN']] disagree for /home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/dpctl/_sycl_core.cpython-36m-x86_64-linux-gnu.so :: 
> WARNING :: get_rpaths_raw()=[] and patchelf=[['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib', '$ORIGIN']] disagree for /home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/dpctl/_memory.cpython-36m-x86_64-linux-gnu.so :: 
> WARNING :: get_rpaths_raw()=['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib'] and patchelf=[['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib']] disagree for /home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/dpctl/libDPPLOpenCLInterface.so :: 
> WARNING :: get_rpaths_raw()=['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib'] and patchelf=[['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib']] disagree for /home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/dpctl/libDPPLSyclInterface.so :: 
> WARNING :: get_rpaths_raw()=['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib'] and patchelf=[['/home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib']] disagree for /home/diptorupd/.conda/envs/dpctl-devel2/conda-bld/dpctl_1601153736731/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.6/site-packages/dpctl/_opencl_core.abi3.so :: 
> Traceback (most recent call last):
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/bin/conda-build", line 11, in <module>
>     sys.exit(main())
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 474, in main
>     execute(sys.argv[1:])
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 465, in execute
>     verify=args.verify, variants=args.variants)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/api.py", line 210, in build
>     notest=notest, variants=variants)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/build.py", line 3017, in build_tree
>     notest=notest,
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/build.py", line 2299, in build
>     newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/build.py", line 1598, in bundle_conda
>     files = post_process_files(metadata, initial_files)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/build.py", line 1444, in post_process_files
>     post_build(m, new_files, build_python=python)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/post.py", line 1315, in post_build
>     check_overlinking(m, files, host_prefix)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/post.py", line 1243, in check_overlinking
>     m.config.variant)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/post.py", line 1058, in check_overlinking_impl
>     for package in packages}
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/post.py", line 1058, in <dictcomp>
>     for package in packages}
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/post.py", line 601, in library_nature
>     dsos, run_exports, _ = determine_package_nature(pkg, prefix, subdir, bldpkgs_dirs, output_folder, channel_urls)
>   File "/home/diptorupd/.conda/envs/dpctl-devel2/lib/python3.6/site-packages/conda_build/post.py", line 583, in determine_package_nature
>     assert isinstance(channeldata, dict) or isinstance(pkg, FakeDist)
> AssertionError

**Comment by PokhodenkoSA** [comment]
@diptorupd it is known issue. Use conda-build=3.18. It does not work with newest conda-build=3.20.

**Comment by oleksandr-pavlyk** [comment]
@PokhodenkoSA Were you able to build conda package with this?

**Comment by PokhodenkoSA** [comment]
@oleksandr-pavlyk I did not try this concrete PR but switching to conda-build=3.18 will eliminate problem described by @diptorupd.

**Comment by oleksandr-pavlyk** [comment]
Trying `conda build conda-recipe/` fails locally as `{{ compiler('cxx')}}` is being interpreted as GCC. How do you tell `conda-build` to build with `dpcpp` instead?

**Comment by diptorupd** [comment]
@PokhodenkoSA @oleksandr-pavlyk Any ideas why CI is failing at the build stage for this PR?

**Comment by PokhodenkoSA** [comment]
> @PokhodenkoSA @oleksandr-pavlyk Any ideas why CI is failing at the build stage for this PR?

In Linux it is infrastructure problem. On Windows build and test passed - only validation checks False positive.
I think we need to restart CI.

**Comment by diptorupd** [comment]
LGTM. Let us merge after @PokhodenkoSA restarts the Linux CI and we get a clean build on Linux.

**Comment by oleksandr-pavlyk** [comment]
Was able to build locally on Linux, merging?

**Pull request merged**